### PR TITLE
Jesse/bug fixes

### PR DESCRIPTION
**Pull request by ChenJesse**
Addresses #224 #244 #222 #239

### CODE REVIEW

File: Tempo/Controllers/CreateUsernameViewController.swift [hunk 1]
```diff
@@ -156,6 +156,8 @@ class CreateUsernameViewController: UIViewController {
 			updateAlert(correct: false, message: "Username can only contain underscores and alphanumeric characters.")
 		} else if username.characters.count > 18 {
 			updateAlert(correct: false, message: "Username exceeds 18 characters limit.")
+		} else if CharacterSet.decimalDigits.contains(username.unicodeScalars.first!) || username.characters.first == "_" {
```
Comment (ChenJesse) [inline Tempo/Controllers/CreateUsernameViewController.swift]: #239

File: Tempo/Controllers/SearchViewController.swift [hunk 2]
```diff
@@ -169,11 +169,14 @@ class SearchViewController: UIViewController, UITableViewDataSource, UITableView
 				didTogglePlaying(animate: true)
 			}
 			
-			cell.shareButton.isHidden = false
-			cell.shareButtonWidthConstraint.constant = shareButtonWidth
+			//hide shareButton for previous cell, if needed
 			selectedCell?.shareButton.isHidden = true
 			selectedCell?.shareButtonWidthConstraint.constant = 0
 			
+			//show shareButton for current cell
```
Comment (ChenJesse) [inline Tempo/Controllers/SearchViewController.swift:176]: #222

File: Tempo/Profile/ProfileViewController.swift [hunk 3]
```diff
@@ -235,8 +235,10 @@ class ProfileViewController: UIViewController, UIViewControllerTransitioningDele
 				self.showErrorAlert("Oh no!", message: "Username must have at least one character.", actionTitle: "Try again")
 			} else if invalidChars != nil {
 				self.showErrorAlert("Invalid characters", message: "Only underscores and alphanumeric characters are allowed.", actionTitle: "Try again")
-			} else if newUsername.characters.count > 18{
```
Comment (ChenJesse) [inline Tempo/Profile/ProfileViewController.swift:238]: #239

File: Tempo/Profile/UsersViewController.swift [hunk 4]
```diff
@@ -108,8 +108,6 @@ class UsersViewController: UIViewController, UITableViewDelegate, UITableViewDat
 	override func viewWillAppear(_ animated: Bool) {
 		super.viewWillAppear(animated)
 		
-		tableView.tableHeaderView = notConnected(true) ? nil : searchController.searchBar
```
Comment (ChenJesse) [inline Tempo/Profile/UsersViewController.swift:111]: #224

File: Tempo/Views/ParentPlayerCellView.swift [hunk 5]
```diff
@@ -17,7 +17,7 @@ class ParentPlayerCellView: UIView {
 	
 	weak var delegate: PlayerDelegate? {
 		didSet {
-			postsLikable = delegate is FeedViewController
```
Comment (ChenJesse) [inline Tempo/Views/ParentPlayerCellView.swift:20]: #244

**Comment by anniexcheng** [comment]
Looks good!

**Pull request merged**